lock-impl/src/main/java/com/palantir/lock/impl/LockServiceImpl.java

```diff
@@ -1241,10 +1243,28 @@ private String safeLockStats() {
     @Override
     public void close() {
         if (isShutDown.compareAndSet(false, true)) {
-            lockReapRunner.close();
-            threadInfoSnapshotManager.close();
-            blockingThreads.forEach(Thread::interrupt);
-            callOnClose.run();
+            List<Pair<String, Runnable>> closures = List.of(
```

Comment:
nit/pedantry: I'd be careful with this name - I think you're using the noun form of `close`, but this does have a slightly different meaning in computer science (a scope in which variables have side-effects preserved)?